src/bakery/Bakery.cpp

```diff
@@ -93,32 +97,108 @@ Time Bakery::nearestNeighbour() {
-                clientVertices.push_back(client.getVertex());
+        for (Client* client : this->clients)
+            if(!client->getVertex()->visited)
+                clientVertices.push_back(client->getVertex());
 
-        this->graph.dijkstraShortestPath(v, clientVertices);
+        Client *closestClient = this->graph.dijkstraClosestClient(v, clientVertices);
+        cout << "Visited " << closestClient->getName()  << endl;
 
-        Client *closestClient;
-        double minDist = INF;
+        v = closestClient->getVertex();
+        cout << "dist: " << v->dist << endl;
 
-        for (Client& client : this->clients)
-            if (client.getVertex()->dist < minDist && (!client.getVertex()->visited)) {
-                minDist = client.getVertex()->dist;
-                closestClient = &client;
-            }
+        //TODO: ADD EDGES AND CLIENTS
+        van.makeDelivery(Time(v->dist), Time(0), closestClient->getBreadQuantity());
+    }
 
-        cout << "Visited " << closestClient->getName()  << endl;
+    // TODO: TIME SHOULDN'T BE DOUBLE
+    double returningTime = this->graph.bidirectionalDijkstra(v, this->startingVertex);
 
-        v = closestClient->getVertex();
-        travelTime += minDist;
+    cout << "dist retorno: " << returningTime << endl;
+
+    van.addTime(Time(returningTime));
+    van.setClients(clients);
+}
+
+Client *Bakery::getClosestClient() {
+    Client* closestClient;
+    double minDist = INF;
+
+    for (Client* client : this->clients)
+        if (client->getVertex()->dist < minDist && (!client->getVertex()->visited)) {
+            minDist = client->getVertex()->dist;
+            closestClient = client;
+        }
+
+    cout << "Visited " << closestClient->getName()  << endl;
+    return closestClient;
+}
+
+void Bakery::filterClients() {
+    vector<Client *>::iterator it;
+    for (it = clients.begin(); it != clients.end(); ++it) {
+        Client* c = *it;
+        if (graph.findVertex(c->getVertex()->getId()) == NULL) {
+            it = clients.erase(it);
+            --it;
+        }
     }
+}
 
-    this->graph.dijkstraShortestPath(v, this->startingVertex);
-    double returningTime = this->startingVertex->getDist();
+void Bakery::solveFirstPhase() {
+    graph.removeUnreachableVertexes(startingVertex, radius);
+    filterClients();
+    Van& v = vans[0];
+    nearestNeighbour(v);
+    cout << "Time: " << v.getTotalTime() << endl << "Bread delivered: " << v.getDeliveredBread() << endl << endl;
+}
 
-    return Time(travelTime + returningTime);
+void Bakery::greedyWithDijkstra(Van& van) {
+    sort(clients.begin(), clients.end(), [](const Client* c1, const Client* c2) -> bool {
+        return c1->getDeliveryTime() < c2->getDeliveryTime();
+    });
+
+    Vertex *v1 = startingVertex, *v2;
+    Time start(7, 0);
+    for (int i = 0; i < clients.size(); ++i) {
+        Client* client = clients[i];
+        v2 = client->getVertex();
+        graph.dijkstraShortestPath(v1, v2);
```

Comment:
This should be done but it's a little tricky since I'm using a function to read the vertices' edges (on their path) and the bidirectional dijkstra isn't setting this parameter right, apparently. We need to fix that and add the edges from the last client to the bakery

I wrote TODOs about this too